src/databricks/labs/ucx/framework/dashboards.py

```diff
@@ -97,36 +99,61 @@ def __init__(
     def _query_state(self):
         return f"{self._remote_folder}/state.json"
 
-    @property
-    def dashboard_link(self):
-        return f"{self._ws.config.host}/sql/dashboards/{self._state['dashboard_id']}"
-
-    def create_dashboard(self) -> str:
-        desired_queries = self._desired_queries()
-        parent = self._installed_query_state()
-        data_source_id = self._dashboard_data_source()
-        self._install_dashboard(parent)
-        for query in desired_queries:
-            self._install_query(query, data_source_id, parent)
-            self._install_viz(query)
-            self._install_widget(query)
-        self._store_query_state(desired_queries)
-        return self._state["dashboard_id"]
+    def dashboard_link(self, dashboard_ref: str):
+        dashboard_id = self._state[f"{dashboard_ref}:dashboard_id"]
+        return f"{self._ws.config.host}/sql/dashboards/{dashboard_id}"
+
+    def create_dashboards(self) -> dict:
+        dashboards = {}
+        queries_per_dashboard = {}
+        # Iterate over dashboards for each step, represented as first-level folders
+        step_folders = [f for f in self._local_folder.glob("*") if f.is_dir()]
+        for step_folder in step_folders:
+            logger.debug(f"Reading step folder {step_folder}...")
+            dashboard_folders = [f for f in step_folder.glob("*") if f.is_dir()]
+            # Create separate dashboards per step, represented as second-level folders
+            for dashboard_folder in dashboard_folders:
+                logger.debug(f"Reading dashboard folder {dashboard_folder}...")
+                main_name = os.path.basename(step_folder).title()
```

Comment:
Per @nfx we rather use pathlib rather than the native os methods for path manipulations file access.
Use our new logging methodology as a reference